bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/scope/binding/BindingItemProviderDelegate.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.automation.jsscripting.internal.scope.binding;
+
+import java.util.Collection;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.common.registry.AbstractProvider;
+import org.openhab.core.common.registry.Provider;
+import org.openhab.core.common.registry.ProviderChangeListener;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.items.Item;
+import org.openhab.core.items.ItemProvider;
+import org.openhab.core.items.Metadata;
+import org.openhab.core.items.MetadataKey;
+import org.openhab.core.model.item.BindingConfigReader;
```

Comment:
An add-on must not have a dependency on model (=DSL) classes. What is this about?
Besides, the "binding config readers" are a remainder from 1.x times... All information provided this way should be made available as item metadata through the core framework, so I'd hope you actually do not need this dependency here (although it looks as if this will require some refactoring...).

Reply:
So i went back through our helper library and removed the code that depended on this, this was a mostly unused portion of the library that was there for historic reasons.   So all of this has been removed from the binding.